shell/apps/chrome-extension/chrome-data.js

```diff
@@ -0,0 +1,65 @@
+/*
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import Xen from '../../components/xen/xen.js';
+
+import {deduplicate, filter, flatten} from './data-processing.js';
+
+class ChromeData extends Xen.Base {
```

Comment:
Add comment on what this class does. Maybe a name like `ChromeMessageReceiver` would make it more obvious?

Reply:
BrowserDataReceiver?